provisioning/service/src/Query.cs

```diff
@@ -97,10 +97,7 @@ public class Query : IDisposable
                 throw new ArgumentException($"{nameof(pageSize)} cannot be negative.");
             }
 
-            // TODO: Refactor ContractApiHttp being created again
-            _contractApiHttp = new ContractApiHttp(
-                serviceConnectionString.HttpsEndpoint,
-                serviceConnectionString, httpTransportSettings);
+            _contractApiHttp = contractApiHttp;
```

Comment:
It looks like it is still being disposed in this class. Should it be?

Reply:
Good catch. It shouldn't be disposed here. It should only be disposed when the service client is disposed.